druntime/src/core/stdc/stdarg.d

```diff
@@ -285,6 +285,60 @@ T va_arg(T)(ref va_list ap)
         ap += T.sizeof.alignUp;
         return *p;
     }
+    else version (WebAssembly)
+    {
+        template getSingleWrappedScalarType(T)
+        {
+            static if (__traits(isScalar,T))
```

Comment:
For LDC, I decided to treat class references and associative arrays as regular pointers, i.e., scalars.